plugins/test/unit/plugins/db/test_models.py

```diff
@@ -80,18 +89,36 @@ def test_rpm_search_dicts_sanitizes_checksum_type_sums(self):
         Assert that the rpm_search_dicts() method properly sanitizes checksum types with the sum
         is specified with the 'type' attribute.
         """
-        errata = models.Errata('id', {})
-        errata.metadata = {
-            'pkglist': [
-                {'packages': [
-                    {'name': 'name', 'epoch': '0', 'version': '0.0', 'sums': ['sum1', 'sum2'],
-                     'release': 'release', 'arch': 'arch', 'type': 'sha'}]}]}
+        errata = models.Errata()
+        errata.pkglist = [
+            {'packages': [
+                {'name': 'name', 'epoch': '0', 'version': '0.0', 'sums': ['sum1', 'sum2'],
+                 'release': 'release', 'arch': 'arch', 'type': 'sha'}]}]
 
         ret = errata.rpm_search_dicts
 
         self.assertEqual(len(ret), 1)
         self.assertEqual(ret[0]['checksumtype'], 'sha1')
 
+    def test_rpm_search_dicts_no_checksum(self):
+        """
+        Assert that the rpm_search_dicts() method tolerates a missing checksumtype, as is found
+        when using this demo repo: https://repos.fedorapeople.org/repos/pulp/pulp/demo_repos/zoo/
+        """
+        errata = models.Errata()
+        errata.pkglist = [
+            {'packages': [
+                {'name': 'foo', 'epoch': '0', 'version': '0.0', 'sum': None,
+                 'release': 'release', 'arch': 'arch'}]}]
+
+        ret = errata.rpm_search_dicts
+
+        # sanity-check that there is one result with the correct name
+        self.assertEqual(len(ret), 1)
+        self.assertEqual(ret[0]['name'], 'foo')
+        # make sure this field is still not present
+        self.assertTrue('checksumtype' not in ret[0])
```

Comment:
To check the negation I usually use assertFalse, but it's a matter of preference.